vividus-bdd-engine/src/main/java/org/vividus/bdd/transformer/FilteringTableTransformer.java

```diff
@@ -57,18 +52,13 @@ public String transform(String tableAsString, TableParsers tableParsers, TablePr
         isTrue(!(byMaxColumns != null && byColumnNames != null),
                 "Conflicting properties declaration found: '%s' and '%s'",
                 BY_MAX_COLUMNS_PROPERTY, BY_COLUMNS_NAMES_PROPERTY);
-        ExamplesTable examplesTable = examplesTableFactory.get().createExamplesTable(tableAsString);
-
-        List<String> filteredColumns = getFilteredHeaders(byMaxColumns, byColumnNames, examplesTable.getHeaders());
-        List<Map<String, String>> result = filterByHeaders(filteredColumns, getFilteredRows(byMaxRows, examplesTable));
+        TableRows tableRows = tableParsers.parseRows(tableAsString, properties);
 
-        List<List<String>> resultRows = result.stream()
-                .map(LinkedHashMap::new)
-                .map(Map::values)
-                .map(ArrayList::new)
-                .collect(Collectors.toList());
+        List<String> filteredColumns = getFilteredHeaders(byMaxColumns, byColumnNames, tableRows.getHeaders());
+        List<Map<String, String>> result = filterByHeaders(filteredColumns,
+                getFilteredRows(byMaxRows, tableRows.getRows()));
 
-        return ExamplesTableProcessor.buildExamplesTable(filteredColumns, resultRows, properties, true, true);
+        return ExamplesTableProcessor.buildExamplesTable(filteredColumns, result, properties);
```

Comment:
why not `ExamplesTableProcessor.buildExamplesTable(filteredColumns, result, properties, true, true);` ?

Reply:
are you suggesting to refactor all the existing classes that rely on List<List<String>>?